### DISCUSSION TITLE
Allow the user to add files

### DISCUSSION THREAD

**Discussion by GaetanRdn** [stackblitz/tutorialkit#204] [Ideas]
Like in Stackblitz, the possibility to add file from the file tree could be a great feature.

We could create bigger exercices, more difficult. 

In my case, I am preparing a tutorial for Angular. And with it, we quickly need to create new files.

**Comment by tadjik1** [comment]
Same here, my course is about node.js, and creating and manipulating files is a must-have feature. Maybe I can contribute somehow and help with the implementation?

**Comment by AriPerkkio** [comment]
Created an issue for this: https://github.com/stackblitz/tutorialkit/issues/263

**Comment by isaacplmann** [comment]
I would like the same feature.

I think the most straightforward way to implement this would be to have the `FileTree` component show the current file system instead of the predefined contents in the `_files` folder.  When that mode is enabled, the file syncing can be turned off.

The terminal does allow you to create and modify files - they just don't show up in the FileTree.  `npm add cowsay && cat package.json` will show the changes.
